account_type_inactive/views/account_type.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_account_type_form" model="ir.ui.view">
+        <field name="name">account.account.type.form.inherit</field>
+        <field name="model">account.account.type</field>
+        <field name="inherit_id" ref="account.view_account_type_form" />
+        <field name="arch" type="xml">
+            <field name="type" position="after">
+                <field name="active" />
```

Comment:
The field type in this view is the 'Internal type' Selection field and is only used to display the active checkbox below it on the form view.
Account types marked as inactive won't be displayed on the tree view or Many2one selectors (default odoo behaviour) anyway.